xcube/core/normalize.py

```diff
@@ -726,6 +722,8 @@ def _normalize_dim_order(ds: xr.Dataset) -> xr.Dataset:
                 ds = ds.copy()
                 copy_created = True
             ds[var_name] = var.transpose(*dim_names)
+            if var.encoding:
+                ds[var_name].encoding['chunks'] = ds[var_name].data.chunksize
```

Comment:
`ds[var_name].data.chunksize` requires that `ds[var_name].data` has an attribute `chunksize` but that cannot be assumed. It is safer to use the xarray API, but only if it is chunked:
```suggestion
            if 'chunks' in var.encoding and ds[var_name].chunks is not None:
                ds[var_name].encoding['chunks'] = ds[var_name].chunks
```

Reply:
Even this might lead to problems because dask chunking is more general than Zarr chunking. So there are dask chunks that cann not be used in Zarr. Zarr requires all but the last chunk to be same size, while dask allows for any chunk sizes greater zero.

Reply:
I'd like to use 

if 'chunksize' in ds[var_name].data: 

then. The reasong behind this change is that ds[var_name].encoding['chunks'] and ds[var_name].data.chunksize should have the same values, which is not correct anymore after the dimension order has changed.
What do you suggest?

Reply:
> if 'chunksize' in ds[var_name].data:

That is definitely wrong because you make the assumption that `data` implements `__contains__()`, which you can't.

Reply:
> which is not correct anymore after the dimension order has changed.

I don't get that. In this case the order of `ds[var_name].chunks` changed too. So why is there a problem?

Reply:
It is required during the determination of the zarr chunks that any given chunking in the encoding corresponds to the chunking of the data sets. See here: https://github.com/pydata/xarray/blob/5735e163bea43ec9bc3c2e640fbf25a1d4a9d0c0/xarray/backends/zarr.py#L159